src/deadline/nuke_submitter/deadline_submitter_for_nuke.py

```diff
@@ -51,6 +51,29 @@ def _get_write_node(settings: RenderSubmitterUISettings) -> tuple[Node, str]:
     return write_node, settings.write_node_selection
 
 
+def _recursive_set_timeouts(
```

Comment:
> Why is this function recursive? 

Since job templates are effectively a tree structure, finding the "actions" within that tree was easiest by recursing through the template. Dynamically finding the actions like this also allows us to modify the job template in the future without being concerned whether we need to change the implementation of this function.

> Do Nuke jobs have multiple render steps?

Right now Nuke jobs only have a single step named `Render` [(link)](https://github.com/aws-deadline/deadline-cloud-for-nuke/blob/mainline/src/deadline/nuke_submitter/default_nuke_job_template.yaml#L61)

Reply:
Crawling the whole job seems unnecessarily complex. We can just directly access the actions with `job_template["steps"][0]["stepEnvironments"][0]["script"]` right? It also makes unintended side effects less likely since we're specifying the part of the template we want to modify instead of any parameter called "actions".

Reply:
>It also makes unintended side effects less likely since we're specifying the part of the template we want to modify instead of any parameter called "actions".

Looking at the current OpenJD spec I don't think that this would be possible because we only look at key names when checking for `action`. Regardless, I didn't want to rule it out completely so I've implemented a non-recursive version of this function which access each action directly. It ended up only being a handful of lines longer and is a more explicit which is nice